File: output/llama3.1-8b/avatar/Python/Java/codeforces_705_B.java
import java.util.Scanner;
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class codeforces_705_B {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        String[] input = br.readLine().split(" ");
        int[] w = new int[input.length];
        for (int i = 0; i < input.length; i++) {
            w[i] = Integer.parseInt(input[i]);
        }
        int c = 2;
        for (int i : w) {
            if (i == 1) {
                System.out.println(c);
            } else if (i % 2 == 0) {
                c = 3 - c;
                System.out.println(c);
            } else {
                System.out.println(c);
            }
        }
    }
}

File: output/llama3.1-8b/avatar/Java/Python/codeforces_705_B.py
import sys
from collections import defaultdict

class FastReader:
    def __init__(self):
        self.br = sys.stdin

    def next(self):
        while not hasattr(self, 'st') or not self.st:
            try:
                self.st = self.br.readline().split()
            except Exception as e:
                print(e)
        return self.st.pop(0)

    def nextInt(self):
        return int(self.next())

    def nextLong(self):
        return int(self.next())

    def nextDouble(self):
        return float(self.next())

    def nextLine(self):
        try:
            return self.br.readline().strip()
        except Exception as e:
            print(e)
            return ""

class Codeforces_705_B:
    scan = FastReader()

    @staticmethod
    def solve():
        t = Codeforces_705_B.scan.nextInt()
        arr = [Codeforces_705_B.scan.nextInt() for _ in range(t)]
        prev_winner = 0

        for i, num in enumerate(arr):
            if num == 1:
                if prev_winner == 0:
                    prev_winner = 2
            if prev_winner == 2 or prev_winner == 0:
                if (num - 1) % 2 == 0:
                    print(2)
                    prev_winner = 2
                else:
                    print(1)
                    prev_winner = 1
            else:
                if (num - 1) % 2 == 0:
                    print(1)
                    prev_winner = 1
                else:
                    print(2)
                    prev_winner = 2

    @staticmethod
    def main():
        t = 1
        while t > 0:
            Codeforces_705_B.solve()
            t -= 1

if __name__ == "__main__":
    Codeforces_705_B.main()